tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/InstanceDiscoveryManagerTests.cs

```diff
@@ -122,6 +127,59 @@ public async Task NetworkCacheProvider_IsUsedFirst_Async()
             Assert.AreSame(_expectedResult, actualResult2, "The static provider should be queried first");
         }
 
+        [TestMethod]
+        public async Task InstanceDiscoveryDisabled_Async()
+        {
+            // Arrange
+            InitializeTestObjects(false);
+            INetworkMetadataProvider networkMetadataProvider = new NetworkMetadataProvider(
+                Substitute.For<IHttpManager>(),
+                _networkCacheMetadataProvider);
+
+            _networkCacheMetadataProvider.GetMetadata("some_env.com", Arg.Any<ILoggerAdapter>()).Returns(_expectedResult);
```

Comment:
Yes, it should not be called, but I put this here because I want to be sure that the cached entry is not returned when the flag is set to false. There is an assert that validates that this is true.